server/src/dependency.rs

```diff
@@ -0,0 +1,169 @@
+use {
+    move_binary_format::errors::PartialVMError,
+    moved::{
+        block::{BlockHash, BlockQueries, BlockRepository, MovedBlockHash},
+        move_execution::{BaseTokenAccounts, MovedBaseTokenAccounts},
+        receipt::{ReceiptQueries, ReceiptRepository},
+        transaction::{TransactionQueries, TransactionRepository},
+    },
+    moved_genesis::config::GenesisConfig,
+    moved_state::State,
+};
+
+#[cfg(feature = "storage")]
+pub type SharedStorage = &'static moved_storage::RocksDb;
+#[cfg(not(feature = "storage"))]
+pub type SharedStorage = moved::in_memory::SharedMemory;
+#[cfg(feature = "storage")]
+pub type ReceiptStorage = &'static moved_storage::RocksDb;
+#[cfg(not(feature = "storage"))]
+pub type ReceiptStorage = moved::receipt::ReceiptMemory;
+#[cfg(feature = "storage")]
+pub type StateQueries = moved::state_actor::InMemoryStateQueries;
+#[cfg(not(feature = "storage"))]
+pub type StateQueries = moved::state_actor::InMemoryStateQueries;
+
+pub fn block_hash() -> impl BlockHash + Send + Sync + 'static {
+    MovedBlockHash
+}
+
+pub fn base_token(
+    genesis_config: &GenesisConfig,
+) -> impl BaseTokenAccounts + Send + Sync + 'static {
+    MovedBaseTokenAccounts::new(genesis_config.treasury)
+}
+
+pub fn memory() -> SharedStorage {
+    #[cfg(feature = "storage")]
```

Comment:
I think we should make the RocksDB storage the default and keep in-memory as an option with feature named `memory`. Going forward the DB will be the default choice anyways.

We should see how the tests are affected by this. If tests are misbehaving with the DB, then we can run tests with the `--features memory` feature.

Reply:
That does make sense. However, from a performance standpoint, the `rocksdb` adds additional dependencies that affect the compilation time significantly.

Turning it on explicitly makes for faster compilation time by default. The `memory` flag would need to be used in conjuction with `--no-default-features` to get that same benefit.

We will probably add a configuration option for the same purpose too, that can alert you if the compiled binary has improper flags set.